### PR TITLE
tt-rss service: Use nginx virtualhosts; improve config options

### PR DESCRIPTION
**Pull request by Nadrieril**
###### Motivation for this change
#15862 has been merged, thus allowing to re-add virtualhost support in tt-rss.

I also fixed some problems with the configuration.
###### Things done
- [x] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/web-apps/tt-rss.nix [hunk 1]
```diff
@@ -466,34 +468,32 @@ let
...
-    #     extraConfig = ''
-    #       access_log  /var/log/nginx-${virtualHostName}-access.log;
-    #       error_log   /var/log/nginx-${virtualHostName}-error.log;
-    #     '';
+    services.nginx.virtualHosts = mkIf (cfg.virtualHost != null) {
+      "${cfg.virtualHost}" = {
+        root = "${cfg.root}";
+        extraConfig = ''
+          access_log  /var/log/nginx-${cfg.virtualHost}-access.log;
+          error_log   /var/log/nginx-${cfg.virtualHost}-error.log;
```
Comment (globin) [inline nixos/modules/services/web-apps/tt-rss.nix]: I'd personally not expect this, as all my other nginx logs go to systemd journal.

File: nixos/modules/services/web-apps/tt-rss.nix [hunk 2]
```diff
@@ -466,34 +468,32 @@ let
...
+        '';
 
-    #     locations."/" = {
-    #       extraConfig = ''
-    #         index index.php;
-    #       '';
-    #     };
+        locations."/" = {
+          extraConfig = ''
+            index index.php;
```
Comment (globin) [inline nixos/modules/services/web-apps/tt-rss.nix]: Can use `index` attr now.

File: nixos/modules/services/web-apps/tt-rss.nix [hunk 3]
```diff
@@ -466,34 +468,32 @@ let
...
+            fastcgi_pass unix:${phpfpmSocketName};
+            fastcgi_index index.php;
+            fastcgi_param SCRIPT_FILENAME ${cfg.root}/$fastcgi_script_name;
 
-    #         include ${pkgs.nginx}/conf/fastcgi_params;
-    #       '';
-    #     };
-    #   };
-    # } else {};
+            include ${pkgs.nginx}/conf/fastcgi_params;
```
Comment (globin) [inline nixos/modules/services/web-apps/tt-rss.nix]: Are included by default since #15862.

File: nixos/modules/services/web-apps/tt-rss.nix (hunk 1 above)
Comment (grahamc) [inline nixos/modules/services/web-apps/tt-rss.nix]: Same.

File: nixos/modules/services/web-apps/tt-rss.nix (hunk 3 above)
Comment (Nadrieril) [inline nixos/modules/services/web-apps/tt-rss.nix]: I can't seem to find it. Is it the same as `${pkgs.nginx}/conf/fastcgi.conf` ?

Comment (Nadrieril) [inline nixos/modules/services/web-apps/tt-rss.nix]: Oh indeed, it's almost the same file. Removing.

**Comment by joachifm** [comment]
Is this ready?

**Comment by Nadrieril** [comment]
As far as I'm concerned, yes

**Pull request merged**